apluslms_roman/backends/docker.py

```diff
@@ -99,20 +103,41 @@ def _run_opts(self, task, step):
 
     def prepare(self, task, observer):
         client = self._client
+        images = self._cache.images
+        day = timedelta(days=1)
         for step in task.steps:
             observer.step_preflight(step)
             image, tag = step.img.split(':', 1)
+
+            last_updated = images.get(step.img, None)
+            last_updated = (day if not last_updated
+                else datetime.now() - last_updated)
```

Comment:
if this okay or would using `last_updated and last_updated >= day` when checking be better?

Reply:
It's kind of hard to read as last_updated is first a datetime, but when it's overwritten, it becomes timedelta. Type changes are not very good for readability and a new variable name should be used.

Reply:
Okay, I can change that. I meant to ask more about the `last_updated = day if not last_updated` part, feels kind of an inelegant way to solve the problem...

Reply:
Hmmm. Logically it kind of makes a problem. If we don't know when it was updated, we should do the update now.

Reply:
there's a `if last_updated >= day` check later, so it's always true that way, but I guess I'll check it to `last_updated and last_updated >= day`

Reply:
Or you can calculate `should_update = not last_updated or last_updated >= day` and then `if not img_found or should_update` (or you can combine it to a long if.

Reply:
Ah that works quite well, thanks